[UX] TypedDict: better error message for KeyError #9892
-------------------------------------------------------

The ``KeyError`` raised from ``__getitem__`` for the ``numba.typed.Dict`` in
case of an non-existing key now reports the value of the key. This behaviour is
in-line with the regular ``dict`` via CPython.

